cardano-node/app/cardano-cli.hs

```diff
@@ -76,21 +84,81 @@ main = do
     renderCliError = show
 
 data CLI = CLI
-  { protocol    :: Protocol
+  { totalConfig :: PartialCardanoConfiguration
   , mainCommand :: ClientCommand
-  , commonCli   :: CommonCLI
-  , commonCliAdv :: CommonCLIAdvanced
-  , loggingCli  :: LoggingCLIArguments
   }
 
 parseClient :: Parser CLI
-parseClient =
-  CLI
-    <$> parseProtocolActual
-    <*> parseClientCommand
-    <*> parseCommonCLI
-    <*> parseCommonCLIAdvanced
-    <*> loggingParser
+parseClient = do
+  ptcl <- (parseProtocolByron <|> parseProtocolRealPBFT)
+  cmd <- parseClientCommand
+  dbPath <- parseDbPath
+  genPath <- parseGenesisPath
+  genHash <- parseGenesisHash
+  delCert <- parseDelegationeCert
+  sKey <- parseSigningKey
+  socketDir <- parseSocketDir
+  pbftSigThresh <- parsePbftSigThreshold
+  reqNetMagic <- parseRequireNetworkMagic
+  slotLength <- parseSlotLength
+  logConfigFp <- lastOption parseLogConfigFile
+  logMetrics <- parseLogMetrics
+  pure $ CLI (createPcc ptcl dbPath genPath genHash delCert sKey socketDir
+                        pbftSigThresh reqNetMagic slotLength logConfigFp
+                        logMetrics)
+              cmd
```

Comment:
Don't you feel that with the `ApplicativeDo`'s introduction of the unnecessary names we've lost some of the clarity here?

I think previously it was much easier on the eyes..

Reply:
@deepfire I'm not opposed to removing `ApplicativeDo`. You have a good point.

Reply:
@deepfire Thinking about it some more, the issue here is then I would need `CLI Protocol FilePath...` etc. `ApplicativeDo` allows me to keep the `CLI` datatype more concise.

Reply:
Just pushed an example of how to do it without any type changes: https://github.com/input-output-hk/cardano-node/commit/a86716311c0d74860140fe61beda91a4565e4da1#diff-deb1de1e23655df9067b46df271864c6L109